### ISSUE TITLE
Sub-menus within the main menu cannot be closed on mobile

### ISSUE THREAD

**Issue by patdahl** [wagtail/wagtail#10747]
### Issue Summary

When using Wagtail on a screen with a viewport narrower than 800, opening a submenu (making a sidebar panel visible) puts the UI into a state where the user cannot return to just viewing the main menu. It appears that the only way to collapse the sidebar panel is to click the sidebar-menu-item__link button, but the is hidden by the sidebar panel itself. The user has to navigate away and return in order to reset the state of the UI so that the main menu is visible again. 

### Steps to Reproduce

1. Using a fresh installation or any project that has submenus in the main menu configuration, view the home cms page on a screen with a viewport width less than 800.
2. Open the main menu.
3. Open the Page explorer (or other submenu, such as Reports).
4. Attempt to close the Page explorer to go back to the main menu.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

-   Using the main menu toggle will close both the sidebar panel and the main menu. However, one can watch via inspector that the class of the sidebar panel does not change. Thus, when the main menu is reopened, the sidebar panel is still open and still covering the main menu.

-   A way to collapse the sidebar panel separate from the main menu toggle would be nice, but I think the expected behavior would be that closing the main menu via the toggle would reset the state of the menu and close and submenus while on mobile. 

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.10.11
-   Django version: 4.2.4
-   Wagtail version: 5.1
-   Browser version:
    - Chrome 115.0.5790.110
    - Edge 115.0.1901.188

**Comment by lb-** [comment]
Confirming this issue, I have encountered this before but did not raise a bug, thank you for raising.

It's pretty frustrating and the only way to close the menu on mobile is to refresh the page / load a different page.

![Screenshot 2023-08-09 at 7 11 16 pm](https://github.com/wagtail/wagtail/assets/1396140/5dc2eae6-43e9-4b6a-aea0-4e8154eabefe)

![Screenshot 2023-08-09 at 7 09 54 pm](https://github.com/wagtail/wagtail/assets/1396140/12e70927-fdf0-4274-a76e-6490ebe2ceab)

**Comment by lb-** [comment]
@thibaudcolas  - Here are some suggestions to fix this (styling only)

Without the main menu close button having access to the state of whether a sub-menu is opened, which would be a bigger JS change, we could move the menu button out/over another part of the content to make it visible.

There may be a different CSS approach to make the sub-menu buttons appear over the close bar that I am not aware of.

## Idea 1. Move the main menu sub-menu link positioned to the top of the sub-menu sidebar.

This may not work well for the page explorer variant though.

![Screenshot 2023-08-09 at 7 39 24 pm](https://github.com/wagtail/wagtail/assets/1396140/0d7fb62b-0ae3-4e78-abf8-50bb2f848558)

## Idea 2. Move the main menu sub-menu link positioned to the bottom of the sub-menu sidebar.

A little confusing for users possibly but should work for most cases, we would need to ensure that it works for the fixed footer sub-menu items like settings.

![Screenshot 2023-08-09 at 7 37 03 pm](https://github.com/wagtail/wagtail/assets/1396140/ad2bc87d-4013-490f-a1f7-26b4e1a4bb85)

**Comment by lb-** [comment]
Thank you @bmihelac & @laymonage  - it's so good to see this fixed.
